### PR TITLE
FINERACT-2081: Fix WholeTerm loan frequency saved on account level

### PR DESCRIPTION
**Pull request by alberto-art3ch**
## Description

WholeTerm loan frequency saved on account level is displayed as PerYear because It is being inherit from Loan Product.

[FINERACT-2081](https://issues.apache.org/jira/browse/FINERACT-2081)

Screenshot 2024-11-04 at 4 25 35 a m

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/AdvancedPaymentAllocationLoanRepaymentScheduleTest.java [hunk 1]
```diff
@@ -5473,6 +5475,45 @@ public void uc149() {
         });
     }
 
+    // UC150: Validate interestRateFrequencyType to set as Whole Term at the Loan account level
```
Comment (adamsaghy) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/AdvancedPaymentAllocationLoanRepaymentScheduleTest.java:5478]: Would you enhance this test case or create a new one which validate the proper interest rate is set and used? I would say validating whether the proper annual interest rate is on the loan and the calculated interest is matching would be sufficient

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly check my comment

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/AdvancedPaymentAllocationLoanRepaymentScheduleTest.java (hunk 1 above)
Comment (alberto-art3ch) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/AdvancedPaymentAllocationLoanRepaymentScheduleTest.java:5478]: Done! Fix to have the right annual interest rate using whole term option applied

**Review by alberto-art3ch — COMMENTED** [review]
Please review this

**Pull request merged**